### ISSUE TITLE
connectors: allow for retrieving OAuth2 client

### ISSUE THREAD

**Issue by srenatus** [coreos/dex#395]
Given an OAuth2 connector (much like the github connector) to an OAuth2 service _that doesn't use proper TLS certificates_ (self-signed or a CA not trusted by the system cert pool), you might have the following bits in the connector config:

```go
type MyConnectorConfig struct {
	ID           string `json:"id"`
	ClientID     string `json:"clientID"`
	ClientSecret string `json:"clientSecret"`
	URL      string `json:"URL"`
	Cert     string `json:"cert"`
	SkipVerify   bool   `json:"insecureSkipVerify"`
}
```

accompanied by

```go
func newMyConnector(clientID, clientSecret, url, cert string, skipVerify bool, cbURL string) (oauth2Connector, error) {
	config := oauth2.Config{
		Credentials: oauth2.ClientCredentials{ID: clientID, Secret: clientSecret},
		AuthURL:     url + myAuthPath,
		TokenURL:    url + myTokenPath,
		RedirectURL: cbURL,
	}

	var client *http.Client
	if chefCert != "" {
		caPool := x509.NewCertPool()
		caPool.AppendCertsFromPEM([]byte(cert))
		tlsConfig := &tls.Config{
			RootCAs: caPool,
		}

		tr := &http.Transport{TLSClientConfig: tlsConfig}
		client = &http.Client{Transport: tr}
	} else if skipVerify {
		tr := &http.Transport{TLSClientConfig: &tls.Config{
			InsecureSkipVerify: true,
		}}
		client = &http.Client{Transport: tr}
	} else {
		client = http.DefaultClient
	}

	cli, err := oauth2.NewClient(client, config)
	if err != nil {
		return nil, err
	}

	return &myOAuth2Connector{
		clientID:     clientID,
		clientSecret: clientSecret,
		client:       cli,
		url:      url,
	}, nil
}
```

This will work fine _until_  the callback handler uses `http.DefaultClient` again [_here_](https://github.com/coreos/dex/blob/master/connector/connector_oauth2.go#L87):

```go
		ident, err := c.conn.Identity(newAuthenticatedClient(token, http.DefaultClient))
		if err != nil {
			log.Errorf("Unable to retrieve identity: %v", err)
```

...at which point TLS verification fails.

**Comment by ericchiang** [comment]
Beyond coreos/go-oidc#70, it also seems reasonable to put an `*http.Client` field directly on the `OIDCConnector` and `OAuth2Connector` structs, configure those fields from the connector config structs, and then use them directly later.

**Comment by ericchiang** [comment]
I believe this is solved now that coreos/go-oidc#70 is in. Please comment if it's not and I'll reopen.
